pwiz_tools/Skyline/Executables/SkylineBatch/SkylineBatch/SkylineBatchConfigForm.cs

```diff
@@ -0,0 +1,303 @@
+﻿/*
+ * Original author: Ali Marsh <alimarsh .at. uw.edu>,
+ *                  MacCoss Lab, Department of Genome Sciences, UW
+ * Copyright 2020 University of Washington - Seattle, WA
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Windows.Forms;
+using SkylineBatch.Properties;
+
+namespace SkylineBatch
+{
+    public enum ConfigAction
+    {
+        Add, Edit, Copy
+    }
+
+    public partial class SkylineBatchConfigForm : Form
+    {
+        // Allows a user to create a new configuration and add it to the list of configurations,
+        // or replace an existing configuration.
+        // Currently running configurations cannot be replaced, and will be opened in a read only mode.
+        
+
+        private readonly IMainUiControl _mainControl;
+        private readonly bool _isBusy;
+        private readonly ConfigAction _action;
+        private readonly DateTime _initialCreated;
+        private readonly List<ReportInfo> _newReportList;
+
+        public SkylineBatchConfigForm(IMainUiControl mainControl, SkylineBatchConfig config, ConfigAction action, bool isBusy)
+        {
+            _action = action;
+            _initialCreated = config?.Created ?? DateTime.MinValue;
+            _newReportList = new List<ReportInfo>();
+
+            _mainControl = mainControl;
+            _isBusy = isBusy;
+            InitializeComponent();

```

Comment:
Unless it's actually necessary to do stuff first, "InitializeComponent" should be the very first line of this constructor.